lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1754,8 +1786,18 @@ def cmd_sessions(*args)
         end
       end
     when 'list', 'list_inactive', nil
+      if search_term
+        matching_sessions = get_matching_sessions(search_term)
+        if matching_sessions.empty?
+          print_error("No matching sessions.")
+          return
+        end
+      end
+      if matching_sessions
+        session_ids = matching_sessions.keys()
+      end
```

Comment:
```suggestion
          return
        else
          session_ids = matching_sessions.keys()
        end
      end
```
You don't need a separate  if statement here, you already know if you have matching sessions or not